src/macaron/repo_finder/commit_finder.py

```diff
@@ -483,6 +459,40 @@ def _build_version_pattern(name: str, version: str) -> tuple[Pattern | None, lis
         return None, [], CommitFinderInfo.REGEX_COMPILE_FAILURE
 
 
+def determine_optional_suffix_index(version: str, parts: list[str]) -> int | None:
+    """Determine optional suffix index of a given version string.
+
+    Version parts that are alphanumeric, and do not come before parts that are purely numeric, can be treated
+    as optional suffixes.
+    E.g.
+    - 1.2.RELEASE -> 'RELEASE' becomes optional.
+    - 3.1.test.2.M5 -> 'M5' becomes optional.
+    Parts that come after a change in seperator are also flagged as optional.
+    - 2.2-3 -> '3' becomes optional.
+    """
```

Comment:
Can you please add the parameters and return value to the docstring?

Also, is the returned index starting from 0 or 1?